cassandra/cluster.py

```diff
@@ -1710,24 +1719,18 @@ class Session(object):
     .. versionadded:: 2.1.0
     """
 
-    client_protocol_handler = ProtocolHandler
-    """
-    Specifies a protocol handler that will be used for client-initiated requests (i.e. no
-    internal driver requests). This can be used to override or extend features such as
-    message or type ser/des.
-
-    The default pure python implementation is :class:`cassandra.protocol.ProtocolHandler`.
-
-    When compiled with Cython, there are also built-in faster alternatives. See :ref:`faster_deser`
-    """
-
+    _client_protocol_handler = ProtocolHandler
+    _protocol_handler = None
```

Comment:
I don't I understand the difference between `_client_protocol_handler` and `_protocol_handler` here. Is there some important distinction being made here, or is this something that could eventually get refactored into a single attribute?

Reply:
In fact, _client_protocol_handler is what's currently in the code. Users can set it to NumpyProtocolHandler when they want to get numpy data rows. With the new code, ProtocolHandler cannot be used anymore without being instantiated with a context and codecs. 

So, we can definitively check if we want to change the API to set another protocol handler (Numpy, Lazy). We can create a follow-up ticket if you want.

Reply:
I've had another look at this, and I think I get it -- so `client_protocol_handler` is an interface for setting and reading the `ProtocolHandler` class we've chosen to hold encoders and decoders. The internal `_protocol_handler` is the instantiated object that actually does the holding.

nit: Would `protocol_handler_class` or `protocol_handler_factory` be a better name for `client_protocol_handler`?

Reply:
Yep, we can refactor this and document the change. The name is indeed confusing. I will go with _class since we are already using that naming scheme at some other places.